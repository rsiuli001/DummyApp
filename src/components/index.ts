export { default as ScreenList } from './ScreenList/ScreenList';
export { default as Buttons } from './Buttons/Buttons';
export { default as Banner } from './Banner/Banner';
