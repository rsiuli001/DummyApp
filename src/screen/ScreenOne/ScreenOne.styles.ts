import { StyleSheet } from 'react-native';
import * as Colors from '../../constants/colors';

const ScreenOneStyle = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
  },
  input: {
    paddingLeft: 10,
    margin: 15,
    height: 40,
    borderWidth: 1,
  },
  userNameContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginRight: 20,
    marginTop: 10,
    marginBottom: 10,
  },
  text: {
    fontWeight: '400',
    fontSize: 16,
  },
  deviceTypeContainer: {
    backgroundColor: Colors.LIGHT_BLUE,
    height: 50,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default ScreenOneStyle;
