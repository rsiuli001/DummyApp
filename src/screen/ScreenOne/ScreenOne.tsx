import React, { FC, useEffect, useState } from 'react';
import { View, Text, TextInput } from 'react-native';
import { connect } from 'react-redux';
import { Banner, ScreenList } from '../../components';
import * as ActionTypes from '../../constants/actionTypes';
import * as ScreenName from '../../constants/screenName';
import * as Strings from '../../constants/strings';
import * as Colors from '../../constants/colors';
import CustomModule from '../../native-modules';
import ScreenOneStyle from './ScreenOne.styles';

interface StateProp {
  userName: string;
}

interface DispatchProp {
  addUser: (userName: string) => any;
}

export const ScreenOne: FC<any> = ({ navigation, userName, addUser }: any): JSX.Element => {
  const [isSimulator, setIsSimulator] = useState(null);

  useEffect(() => {
    CustomModule.getDeviceType((deviceType: any) => {
      setIsSimulator(deviceType);
    });
  }, []);

  const onPress = (screenName: string) => {
    navigation.navigate(screenName);
  };

  return (
    <View style={ScreenOneStyle.container}>
      <Banner deviceType={isSimulator || false} styles={ScreenOneStyle.deviceTypeContainer} />
      <View style={ScreenOneStyle.userNameContainer}>
        <Text style={ScreenOneStyle.text}>
          {Strings.WELCOME} {userName}
        </Text>
      </View>

      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_ONE}`}
        color={Colors.BLACK}
        onPress={onPress}
        screenName={ScreenName.SCREEN_ONE}
      />
      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_TWO}`}
        color={Colors.BLACK}
        onPress={onPress}
        screenName={ScreenName.SCREEN_TWO}
      />
      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_THREE}`}
        color={Colors.BLACK}
        onPress={onPress}
        screenName={ScreenName.SCREEN_THREE}
      />

      <TextInput
        style={ScreenOneStyle.input}
        value={userName}
        onChangeText={(text) => addUser(text)}
        placeholder={Strings.ENTER_NAME}
      />
    </View>
  );
};

const mapStateToProps = ({ userName }: StateProp) => {
  return {
    userName,
  };
};

const mapDispatchToProps = (dispatch: any): DispatchProp => {
  return {
    addUser: (userName: string) => dispatch({ type: ActionTypes.ADD_USER, payload: { userName } }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ScreenOne);
