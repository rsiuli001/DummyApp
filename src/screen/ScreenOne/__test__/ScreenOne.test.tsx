import React from 'react';
import { shallow } from 'enzyme';
import { TextInput, View } from 'react-native';
import { ScreenOne } from '../ScreenOne';
import { Banner, ScreenList } from '../../../components';

describe('Screen one Tests', () => {
  describe('rendering screen one', () => {
    const wrapper = shallow(<ScreenOne />);

    it('should render view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(View).length).toEqual(2);
    });

    it('should render Banner component', () => {
      expect(wrapper.find(Banner).length).toEqual(1);
    });

    it('should render ScreenList component', () => {
      expect(wrapper.find(ScreenList).length).toEqual(3);
    });

    it('should render TextInput component', () => {
      expect(wrapper.find(TextInput).length).toEqual(1);
    });
  });
});
