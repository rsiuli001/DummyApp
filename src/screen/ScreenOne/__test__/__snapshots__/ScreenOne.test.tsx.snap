// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen one Tests rendering screen one should render view 1`] = `ShallowWrapper {}`;
